internal/discord/commands.go

```diff
@@ -10,7 +10,94 @@ import (
 //	user: 	User
 //	reason: string
 func (d *Discord) Kick(s *discordgo.Session, i *discordgo.InteractionCreate) {
-	return
+	dmFailed := false
+	optionMap := mapOptions(i)
+
+	// Log usage of command
+	logMsg, err := d.LogCommand(i.Interaction)
+	if err != nil {
+		fmt.Printf("Failed to log: %v\n", err)
+	}
+
+	userToKick := optionMap["user"].UserValue(nil).ID
+
+	// Check if user exists in guild
+	err = d.CheckUserInGuild(i.GuildID, userToKick)
+	if err != nil {
+		tempstr := fmt.Sprintf("Could not kick user <@%v>", userToKick)
+		fmt.Printf("%v: %v\n", tempstr, err)
+
+		err = StartInteraction(s, i.Interaction, tempstr)
+		if err != nil {
+			fmt.Printf("Unable to send ephemeral message: %v\n", err)
+		}
+
+		return
+	}
+
+	// DM the user regarding the kick
+	channel, err := s.UserChannelCreate(userToKick)
+	if err != nil {
+		tempstr := fmt.Sprintf("Could not create a DM with user %v", userToKick)
+		fmt.Printf("%v: %v\n", tempstr, err)
+		dmFailed = true
+	} else {
+		tempstr := fmt.Sprintf("You are being kicked from `%v` for the following reason:\n%v",
+			GuildName,
+			optionMap["reason"].StringValue(),
+		)
+
+		_, err = s.ChannelMessageSend(channel.ID, tempstr)
+		if err != nil {
+			tempstr := fmt.Sprintf("Could not send a DM to user %v", userToKick)
+			fmt.Printf("%v: %v\n", tempstr, err)
+			dmFailed = true
+		}
+	}
+
+	// Attempt to kick user
+	if len(optionMap["reason"].StringValue()) > 0 {
+		err = d.Session.GuildMemberDeleteWithReason(i.GuildID, userToKick, optionMap["reason"].StringValue())
+	} else {
+		err = StartInteraction(s, i.Interaction, "Please provide a reason for the kick.")
+		if err != nil {
+			fmt.Printf("Unable to send ephemeral message: %v\n", err)
+		}
+
+		return
+	}
```

Comment:
i think we would rather kick the user first before sending the DM. if the kick fails but the person gets the DM, it would look really confusing to the person

Reply:
The reason why I didn't do this was cause I'm pretty sure if I attempted to send the DM after the kick, the bot wouldn't share any servers with the kicked user and wouldn't be able to DM them.
I tried anyway and this is what I got:
![image](https://github.com/naurffxiv/moddingway/assets/41562938/7520f1d0-1475-4a0f-8985-bae4b1ef1edd)

Reply:
ugh thats annoying. ok let's update the error message in the case where we can't kick, the mod should do it manually and _not_ rerun the command. that will send the person two dms which isnt what we want